src/persist/src/indexed/unsealed.rs

```diff
@@ -291,10 +296,13 @@ impl Unsealed {
                 .recv()?
                 .updates
                 .iter()
+                .flat_map(|updates| updates.iter())
                 .filter(|(_, ts, _)| self.ts_lower.less_equal(ts))
-                .cloned(),
+                .map(|((k, v), t, d)| ((k.to_vec(), v.to_vec()), t, d)),
         );
+
         debug_assert!(!updates.is_empty());
+        let updates = vec![updates.iter().collect::<ColumnarRecords>()];
```

Comment:
shouldn't this build the ColumnarRecords directly?

Reply:
I'm sorry I'm not sure I understand the question here

Reply:
oh i think i see the question -- right now we go through a vec to generate the ColumnarRecords and we could skip that intermediate step. yes. ofc

Reply:
This methods builds a `Vec<((Vec<u8>, Vec<u8>), u64, size)>` and then uses that to construct a ColumnarRecords. instead of `let mut updates = vec![]` above, we could that updates variable a ColumnarRecordsBuilder and then instead of `updates.extend`, we could call `updates.push`. this line then becomes `updates.finish` instead of `updates.iter().collect::<ColumnarRecords>()`

Reply:
looks like we raced! yup, that's it exactly

Reply:
done! implemented it with a `FromIterator` implementation so it can be reused if needed (I think its essentially the same builder logic that i would have had to inline). This removes the allocation for the vec, and also removes the allocations for each key and val, so thank you for the suggestion!